src/charm.py

```diff
@@ -184,6 +178,37 @@ def main(self, event):
 
         self.model.unit.status = ActiveStatus()
 
+    def _send_info(self, event, interfaces):
+        if interfaces["kubeflow-profiles"]:
+            interfaces["kubeflow-profiles"].send_data(
+                {
+                    "service-name": self.model.app.name,
+                    "service-port": str(self.model.config["port"]),
+                }
+            )
+
+    def _check_leader(self):
+        if not self.unit.is_leader():
+            # We can't do anything useful when not the leader, so do nothing.
+            raise CheckFailed("Waiting for leadership", WaitingStatus)
+
+    def _get_interfaces(self):
+        try:
+            interfaces = get_interfaces(self)
+        except NoVersionsListed as err:
+            raise CheckFailed(err, WaitingStatus)
+        except NoCompatibleVersions as err:
+            raise CheckFailed(err, BlockedStatus)
+        return interfaces
+
+    def _check_image_details(self):
+        try:
+            profile_image_details = self.profile_image.fetch()
+            kfam_image_details = self.kfam_image.fetch()
+        except OCIImageResourceError as e:
+            raise CheckFailed(f"{e.status.message}", e.status_type)
+        return profile_image_details, kfam_image_details
```

Comment:
Same comment as above.  This isn't needed at all because OCIImageResourceError has a `.status`  We can remove this and just do the following in main:
```
def main():
  try:
    ...
    profile_image_details = self.profile_image.fetch()
    kfam_image_details = self.kfam_image.fetch()
  except CheckFailed, OCIImageResourceError as e:
    self.model.unit.status = e.status
```

Reply:
Although the logic remains similar, piling on exceptions in an `except` statement isn't really nice pattern. It also leaves us with what I mentioned in another comment about abstracting away the status setting to where these exceptions are defined.